src/config.h

```diff
@@ -74,6 +74,7 @@ struct Config {
     std::vector<NamedString> config_entries;  // Entries go into OrtSessionOptions::AddConfigEntry
 
     std::vector<ProviderOptions> provider_options;
+    std::vector<ProviderOptions> provider_options_backup;
```

Comment:
I think my main confusion is calling it 'provider_options_backup' it's not really a backup, it's really the provider options. And we've now separated the options from which provider is currently in use.

Is there a way to not duplicate the options but instead add a new field 'providers' that's just an array of names in provider priority order? We can generate this on load with existing configs that omit it based on the provider options they specify.

Reply:
Sounds good. I'll make this update.